src/extractor/guidance/intersection_handler.cpp

```diff
@@ -71,17 +71,19 @@ TurnType::Enum IntersectionHandler::findBasicTurnType(const EdgeID via_edge,
     if (!on_ramp && onto_ramp)
         return TurnType::OnRamp;
 
-    const auto &in_name =
+    const auto &in_name_id =
         node_data_container.GetAnnotation(node_based_graph.GetEdgeData(via_edge).annotation_data)
             .name_id;
-    const auto &out_name =
+    const auto &out_name_id =
         node_data_container.GetAnnotation(node_based_graph.GetEdgeData(road.eid).annotation_data)
             .name_id;
+    const auto &in_name_empty = name_table.GetNameForID(in_name_id).to_string().empty();
+    const auto &out_name_empty = name_table.GetNameForID(out_name_id).to_string().empty();
```

Comment:
`GetXForID` returns

http://www.boost.org/doc/libs/1_60_0/libs/utility/doc/html/string_ref.html

a zero-copy view onto the data. If you call `.to_string()` on it you make a copy of the data and allocate memory for that. But we only want to know if the data is empty. Check the string ref docs, there is already an `empty` function on the string ref.

Reply:
🙇‍♀️